### PR TITLE
Device factory tests

### PR DESCRIPTION
**Pull request by kvish**
give feedback on tests

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kvish__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            DeviceModel d = DeviceSchemaHelper.BuildDeviceStructure("test", true, null);
+
+            Assert.NotNull(d);
+            Assert.Equal("test", d.DeviceProperties.DeviceID);
+            Assert.Equal("normal", d.DeviceProperties.DeviceState);
+            Assert.Equal(null, d.DeviceProperties.HubEnabledState);
+
+        }
+
+        ~SampleDeviceFactoryTests() { }
```
Comment (Inspiravetion) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: ```~``` ?

**Comment by Inspiravetion** [comment]
lgtm

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.BusinessLogic;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.Common
+{
+    public class SampleDeviceFactoryTests
+    {
+        private ISecurityKeyGenerator _securityKeyGeneratorMock;
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: `Mock<ISecurityKeyGenerator>`

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        public SampleDeviceFactoryTests()
+        {
+            _securityKeyGeneratorMock = (new Mock<ISecurityKeyGenerator>()).Object;
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: remove .Object

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+
+
+        [Fact]
+        public void TestGetSampleDevice()
+        {
+          
+            Random randomnumber = new Random();
+            SecurityKeys keys = _securityKeyGeneratorMock.CreateRandomKeys();
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: you have to mock the CreateRandomKeys method

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        }
+
+
+
+        [Fact]
+        public void TestGetSampleDevice()
+        {
+          
+            Random randomnumber = new Random();
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: use camelCase

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            Assert.NotNull(d);
+            Assert.NotNull(d.DeviceProperties);
+            Assert.NotNull(d.DeviceProperties.DeviceID);
+        }
+
+        [Fact]
+        public void TestGetDefaultDeviceNames()
+        {
+            List<String> s = SampleDeviceFactory.GetDefaultDeviceNames();
+            Assert.NotNull(s);
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: `Assert.NotEmpty(s)`

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.BusinessLogic;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.Common
+{
+    public class SampleDeviceFactoryTests
+    {
+        private ISecurityKeyGenerator _securityKeyGenerator;
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: no need to have it as a data member, just use `new SecurityKeyGenerator()` wherever required.

File: DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            DeviceModel d = DeviceSchemaHelper.BuildDeviceStructure("test", true, null);
+
+            Assert.NotNull(d);
+            Assert.Equal("test", d.DeviceProperties.DeviceID);
+            Assert.Equal("normal", d.DeviceProperties.DeviceState);
+            Assert.Equal(null, d.DeviceProperties.HubEnabledState);
+
+        }
+
+
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure.UnitTests/Common/SampleDeviceFactoryTests.cs]: nit: can you remove extra lines? you can use resharper's code cleanup: ctrl+alt+f

**Pull request merged**